lib/fs.js

```diff
@@ -428,7 +429,7 @@ function tryReadSync(fd, isUserFd, buffer, pos, len) {
 function readFileSync(path, options) {
   options = getOptions(options, { flag: 'r' });
 
-  if (options.encoding === 'utf8' || options.encoding === 'utf-8') {
+  if (normalizeEncoding(options.encoding) === 'utf8') {
```

Comment:
This breaks default encoding

```suggestion
  if (options.encoding && normalizeEncoding(options.encoding) === 'utf8') {
```